### PR TITLE
Fix up CCG related metrics

### PR DESCRIPTION
**Pull request by christarazi**
- metrics: Fix up CCG related metrics
- docs: Update metrics and upgrade reference for CCG metrics

### CODE REVIEW

File: Documentation/operations/upgrade.rst [hunk 1]
```diff
@@ -356,6 +356,10 @@ Changed Metrics
   ``*_api_limiter_wait_duration_seconds`` metrics.
 * The ``cilium_bpf_map_pressure`` for policy maps is now exposed as a single
   label ``cilium_policy_*``, rather than a label per policy map of an endpoint.
+* The ``cilium_cidrgroup_policies`` has been renamed to
+  ``cilium_cidrgroups_referenced`` for better clarity.
+* The ``cilium_idrgroup_translation_time_stats_seconds`` has been disabled by
```
Comment (tommyp1ckles) [inline Documentation/operations/upgrade.rst]: typo idrgroup -> cidrgroup

**Review by tommyp1ckles — CHANGES_REQUESTED** [review]

File: Documentation/operations/upgrade.rst [hunk 2]
```diff
@@ -356,6 +356,10 @@ Changed Metrics
...
+  default.
```
Comment (learnitall) [inline Documentation/operations/upgrade.rst]: Would it be possible to explain why these changes occurred? Why was `cilium_cidrgroup_translation_time_stats_seconds` disabled by default? Why is `cilium_cidrgroup_referenced` better than `cilium_cidrgroup_policies`?

File: Documentation/operations/upgrade.rst [hunk 3]
```diff
@@ -356,6 +356,10 @@ Changed Metrics
   ``*_api_limiter_wait_duration_seconds`` metrics.
 * The ``cilium_bpf_map_pressure`` for policy maps is now exposed as a single
   label ``cilium_policy_*``, rather than a label per policy map of an endpoint.
+* The ``cilium_cidrgroup_policies`` has been renamed to
```
Comment (learnitall) [inline Documentation/operations/upgrade.rst]: Need to add `metric` as the noun here:

```suggestion
* The ``cilium_cidrgroup_policies`` metric has been renamed to
```

**Review by learnitall — CHANGES_REQUESTED** [review]

File: Documentation/operations/upgrade.rst (hunk 2 above)
Comment (christarazi) [reply Documentation/operations/upgrade.rst]: This is explained in the first commit. I don't think it's really necessary to include the rationale also in the upgrade guide. The other notes in this section also don't explain why the changes were made. Ultimately, the main point of the guide is to inform users of changes. For more details, they can dig into the PRs, read changelog, etc.

**Review by learnitall — APPROVED** [review]
LGTM

**Review by aditighag — APPROVED** [review]

**Review by tommyp1ckles — APPROVED** [review]
